### ISSUE TITLE
Requesting help preparing YFCC15M

### ISSUE THREAD

**Issue by ankitkv** [mlfoundations/open_clip#468]
Are there any guides to prepare the YFCC15M (OpenAI subset) dataset for use with open_clip? I've tried to follow the steps [here](https://github.com/NVlabs/GroupViT/blob/main/README.md#yfcc14m), but open_clip seems to hang at the start of training with these shards.

**Comment by ankitkv** [comment]
No, cc3m and cc12m downloaded via img2dataset work fine

**Comment by mitchellnw** [comment]
I'm not really familiar with the prep instructions you've shared.. do the shards seem in the same format of cc12m?

**Comment by ankitkv** [comment]
I am investigating further. However, can you (or anyone) please share the steps that you have followed to prep this dataset?

**Comment by mitchellnw** [comment]
it could be due to https://github.com/mlfoundations/open_clip/issues/278 -- are you using an up to date version?

**Comment by mitchellnw** [comment]
I believe for our initial processing we 1) used this https://github.com/openai/CLIP/blob/main/data/yfcc100m.md 2) then used tarp https://github.com/webdataset/tarp . Unfortunately we've lost the exact commands.

**Comment by ankitkv** [comment]
I was able to prepare the dataset by following the steps [here](https://github.com/NVlabs/GroupViT/blob/main/README.md#yfcc14m) except for the last step, where they convert the dataset into the webdataset format. I just extracted the downloaded zips, modified the tsv file to change the filenames to URLs with "file:///<path>", and then used img2dataset like I would with cc12m. Small detail, but I also needed to remove all carriage returns from the tsv for img2dataset to work properly.
